spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -0,0 +1,192 @@
+
+      RewriteDataFiles.Result result = action.execute();
+
+      return toOutputRows(result);
+    });
+  }
+
+  private RewriteDataFiles checkAndApplyFilter(RewriteDataFiles action, String where) {
+    if (where != null) {
+      ParserInterface sqlParser = spark().sessionState().sqlParser();
+      try {
+        Expression expression = sqlParser.parseExpression(where);
+        return action.filter(SparkExpressionConverter.convertToIcebergExpression(expression));
+      } catch (ParseException e) {
+        throw new IllegalArgumentException("Cannot parse predicates in where option: " + where);
+      }
+    }
+    return action;
+  }
+
+  private RewriteDataFiles checkAndApplyOptions(InternalRow args, RewriteDataFiles action) {
+    Map<String, String> options = Maps.newHashMap();
+    args.getMap(3).foreach(DataTypes.StringType, DataTypes.StringType,
+            (k, v) -> {
+              options.put(k.toString(), v.toString());
+              return BoxedUnit.UNIT;
+            });
+    return action.options(options);
+  }
+
+  private RewriteDataFiles checkAndApplyStrategy(RewriteDataFiles action, String strategy, SortOrder sortOrder) {
+    if (strategy != null) {
+      if (strategy.equalsIgnoreCase("binpack")) {
+        RewriteDataFiles rewriteDataFiles = action.binPack();
+        if (sortOrder != null) {
+          // calling below method to throw the error as user has set both binpack strategy and sortorder
+          return rewriteDataFiles.sort(sortOrder);
+        }
+        return rewriteDataFiles;
+      } else if (strategy.equalsIgnoreCase("sort")) {
+        if (sortOrder == null) {
+          return action.sort();
+        }
+        return action.sort(sortOrder);
+      } else {
+        throw new IllegalArgumentException("unsupported strategy: " + strategy + ". Only binpack,sort is supported");
+      }
+    } else {
+      // override default binpack to sort, as strategy is not set but sort order is set.
+      if (sortOrder != null) {
+        return action.sort(sortOrder);
+      }
+    }
+    return action;
+  }
+
+  private SortOrder collectSortOrders(Table table, String sortOrderStr) {
+    if (sortOrderStr != null) {
+      String prefix = "ALTER TABLE temp WRITE ORDERED BY ";
+      try {
+        // Note: Reusing the existing Iceberg sql parser to avoid implementing the custom parser for sort orders.
+        // To reuse the existing parser, adding a prefix of "ALTER TABLE temp WRITE ORDERED BY"
+        // along with input sort order and parsing it as a plan to collect the sortOrder.
+        LogicalPlan logicalPlan = spark().sessionState().sqlParser().parsePlan(prefix + sortOrderStr);
+        return (new SortOrderParserUtil()).collectSortOrder(
+            table.schema(),
+            ((SetWriteDistributionAndOrdering) logicalPlan).sortOrder());
+      } catch (AnalysisException ex) {
+        throw new IllegalArgumentException("Unable to parse sortOrder: " + ex.getMessage());
```

Comment:
I don't think we want to rethrow the message here since it will be specific to the "Alter Table" statement and that seems like it may be confusing, Maybe just return the "SortOrder" string.

Reply:
agree.